File: src/com/github/jstN0body/adventofcode/python/day2/Dive.py
import sys

inputData = sys.argv
inputData.remove(inputData[0])

depth = 0
dist = 0
for i in range(len(inputData)):
    if i + 1 == len(inputData):
        continue
    action = inputData[i]
    i += 1
    amount = inputData[i]
    if action == "forward":
        dist += int(amount)
    elif action == "up":
        depth -= int(amount)
    elif action == "down":
        depth += int(amount)

print("Depth: {}".format(depth))
print("Dist: {}".format(dist))
print("Answer: {}\n".format(depth * dist))

depth = 0
dist = 0
aim = 0

for i in range(len(inputData)):
    if i + 1 == len(inputData):
        continue
    action = inputData[i]
    i += 1
    amount = inputData[i]
    if action == "forward":
        dist += int(amount)
        depth += int(amount) * aim
    elif action == "up":
        aim -= int(amount)
    elif action == "down":
        aim += int(amount)

print("Depth: {}".format(depth))
print("Dist: {}".format(dist))
print("Answer: {}".format(depth * dist))

File: src/com/github/jstN0body/adventofcode/java/day2/Dive.java
package com.github.jstN0body.adventofcode.java.day2;

public class Dive {
    public static void main(String[] args) {
        int horizontalLoc = 0, depth = 0, aim = 0;
        for (int i = 1; i < args.length; i+= 2) {
            switch (args[i-1]) {
                case "forward":
                    horizontalLoc += Integer.parseInt(args[i]);
                    break;
                case "up":
                    depth -= Integer.parseInt(args[i]);
                    break;
                case "down":
                    depth += Integer.parseInt(args[i]);
            }
        }

        System.out.println(horizontalLoc + " horiz.");
        System.out.println(depth + " depth");
        System.out.println(horizontalLoc * depth + " final answer\n");

        horizontalLoc = 0;
        depth = 0;
        for (int i = 1; i < args.length; i+= 2) {
            switch (args[i-1]) {
                case "forward":
                    horizontalLoc += Integer.parseInt(args[i]);
                    depth += Integer.parseInt(args[i]) * aim;
                    break;
                case "up":
                    aim -= Integer.parseInt(args[i]);
                    break;
                case "down":
                    aim += Integer.parseInt(args[i]);
            }
        }

        System.out.println(horizontalLoc + " horiz.");
        System.out.println(depth + " depth");
        System.out.println(horizontalLoc * depth + " final answer");
    }
}
